src/ui/window.c

```diff
@@ -1625,10 +1619,15 @@ win_print_outgoing_with_receipt(ProfWin* window, const char* show_char, const ch
     receipt->received = FALSE;
 
     const char* myjid = connection_get_fulljid();
+    gboolean corrected = FALSE;
     if (replace_id) {
-        _win_correct(window, message, id, replace_id, myjid);
-        free(receipt); // TODO: probably we should use this in _win_correct()
-    } else {
+        corrected = _win_correct(window, message, id, replace_id, myjid);
```

Comment:
Why isn't `PREF_CORRECTION_ALLOW` checked here?

Reply:
Same as above.

Reply:
I will mark it as "resolved" and once the decision is made on another similar comment, I will apply the change at all places.